src/utils/hooks/api/loans/use-get-account-loans-overview.tsx

```diff
@@ -0,0 +1,180 @@
+import { BorrowPosition, CurrencyExt, CurrencyIdLiteral, LendPosition, newAccountId } from '@interlay/interbtc-api';
+import { MonetaryAmount } from '@interlay/monetary-js';
+import Big from 'big.js';
+import { useCallback } from 'react';
+
+import { convertMonetaryAmountToValueInUSD } from '@/common/utils/utils';
+import { useSubstrateSecureState } from '@/lib/substrate';
+import { getTokenPrice } from '@/utils/helpers/prices';
+
+import { useGetPrices } from '../use-get-prices';
+import { getTotalEarnedInterestUSDValue, getTotalUSDValueOfPositions } from './get-usd-values';
+import { useGetAccountPositions } from './use-get-account-positions';
+
+interface AccountLoansOverview {
+  lendPositions: LendPosition[] | undefined;
+  borrowPositions: BorrowPosition[] | undefined;
+  lentAssetsUSDValue: Big | undefined;
+  totalEarnedInterestUSDValue: Big | undefined;
+  borrowedAssetsUSDValue: Big | undefined;
+  borrowLimitUSDValue: Big | undefined;
+  collateralRatio: Big | undefined;
+  getNewCollateralRatio: (
+    type: 'withdraw' | 'deposit',
+    currency: CurrencyExt,
+    amount: MonetaryAmount<CurrencyExt>
+  ) => Big | undefined;
+  getNewBorrowLimitUSDValue: (
+    type: 'withdraw' | 'deposit',
+    currency: CurrencyExt,
+    amount: MonetaryAmount<CurrencyExt>
+  ) => Big | undefined;
+  getMaxBorrowableAmount: (
+    currency: CurrencyExt,
+    availableCapacity: MonetaryAmount<CurrencyExt>
+  ) => MonetaryAmount<CurrencyExt> | undefined;
+}
+
+const useGetAccountLoansOverview = (): AccountLoansOverview => {
+  const { selectedAccount } = useSubstrateSecureState();
+
+  const accountId = selectedAccount && newAccountId(window.bridge.api, selectedAccount.address);
+  const prices = useGetPrices();
+  const { lendPositions, borrowPositions } = useGetAccountPositions(accountId);
+
+  let lentAssetsUSDValue: Big | undefined = undefined;
```

Comment:
Are all these `let`'s really necessary? For what I can see, it can set these variables normally, without any mutations.

Reply:
There are situation where using `let`/mutations is good, but  generally I try to avoid mutations because it allows a variable to be changed anywhere from inside the scope. Therefore bringing more complexity into the code.

Reply:
@danielsimao Yes, I think in this case usage of `let` is justified by not having too long ternary operator expressions -- those would make code much harder to read imo. Or do you have any other suggestion? :)